diff -ur o-amrita-1.0.2/lib/amrita/xml.rb amrita-1.0.2/lib/amrita/xml.rb
--- o-amrita-1.0.2/lib/amrita/xml.rb	2002-10-22 02:10:31.000000000 -0600
+++ amrita-1.0.2/lib/amrita/xml.rb	2005-03-07 23:54:48.795316417 -0700
@@ -98,7 +98,7 @@
 
     def XMLParser.parse_file(fname, dummy=nil, &block)
       l = Listener.new(&block) 
-      REXML::Document.parse_stream(REXML::File.new(fname), l)
+      REXML::Document.parse_stream(File.new(fname), l)
       l.result
     end
   end
